### DISCUSSION TITLE
Can privileged_session_max_age be removed?

### DISCUSSION THREAD

**Discussion by jonnyijapan** [ory/kratos#2570] [Q&A] [answered]
I'm totally understanding the reason for using privileged_session_max_age - but I just want to confirm:

Can the privileged_session_max_age setting be removed?

https://www.ory.sh/docs/guides/session-management/overview#privileged-sessions

Or maybe there is another way around this, such that there is essentially no limit to the session length? Maybe the session can be extended silently on the backend if needed, would this make the session age "younger"?

**Comment by aeneasr** [comment, marked as answer]
Just set this to a very high value, like a year or so :)

**Reply by jonnyijapan** [reply]
Thanks. I also figured that the normal UI flow would be to require the user to first input the current password, as well as the new password. Then in the backend I'll sign in the user first to get a fresh session token, and use that token to change the password using the selfservice APIs.

**Reply by jonnyijapan** [reply]
@aeneasr 

I realized I need to follow this up.

So the background reason why I want to remove this privileged_session_max_age:

In the Recover account flow, I am using the session token from when the user logged in most recently. I am using that session token to change the password for the user. I use these functions in the Go SDK:

- V0alpha2Api.InitializeSelfServiceSettingsFlowWithoutBrowserExecute
- V0alpha2Api.SubmitSelfServiceSettingsFlow + SubmitSelfServiceSettingsFlowWithPasswordMethodBodyAsSubmitSelfServiceSettingsFlowBody

This requires that the session token is not too old - thus related to privileged_session_max_age.

If privileged_session_max_age is set to a year, and the user forgot the password and tries to restore the account more than a year after logging in the last time, the token stored in our backend cannot change the password for the user.

So I would really need to either:

- Completely remove the privileged_session_max_age limit
- or, use a session cookie (not the token...) to change password for the user, using the Go SDK. Explanation: In the recover account flow, I also have been able to get hold of that temporary (I think it is temporary) session cookie by calling the "self-service/recovery?flow=%s&token=%s" in our backend during the recovery flow. Using that session cookie I have been able to confirm the session for the user such that I know the right user authenticated. What is left here is if there were a way to change the password using only this session cookie, but I have failed doing so.

Here is the complete code I use to change password for a user, using the session token. Can this be rewritten such that it can take a session cookie instead of the session token, and change the password for the user? Maybe it would need some more data, like a CSRF cookie?

```
func changePassword(newPassword string, sessionToken string) bool {
	// From here: https://www.ory.sh/docs/kratos/self-service/flows/user-settings#code-examples-for-nodejs-reactjs-go-
	success := false
	apiClient := getPublicApiClient()

	flow, _, err1 := apiClient.V0alpha2Api.InitializeSelfServiceSettingsFlowWithoutBrowserExecute(kratosclient.V0alpha2ApiApiInitializeSelfServiceSettingsFlowWithoutBrowserRequest{}.XSessionToken(sessionToken))
	if err1 != nil {
		log.Println("(token) Failed InitializeSelfServiceSettingsFlowWithoutBrowserExecute: ", err1)
	} else {
		ctx := context.Background()

		// Submit the form
		_, response, err := apiClient.V0alpha2Api.SubmitSelfServiceSettingsFlow(ctx).Flow(flow.Id).XSessionToken(sessionToken).SubmitSelfServiceSettingsFlowBody(
			kratosclient.SubmitSelfServiceSettingsFlowWithPasswordMethodBodyAsSubmitSelfServiceSettingsFlowBody(&kratosclient.SubmitSelfServiceSettingsFlowWithPasswordMethodBody{
				Method:   "password",
				Password: newPassword,
			}),
		).Execute()
		if err != nil {
			log.Println("Failed SubmitSelfServiceSettingsFlow: ", err)
			log.Println("response:", response)
		} else {
			success = true
		}
	}

	return success
}
```

**Reply by vinckr** [reply]
Hello @jonnyijapan 
just stumpled upon this thread and was wondering if you found a solution in the meantime?
I marked it as answered for now but feel free to revisit at any time.

**Reply by jonnyijapan** [reply]
The related project I worked on came pretty far but unfortunately never saw the light of day. It's been over a year so my memory is vague but I don't think I was able to do it the way I wanted. 

Since I no longer need this I guess closing this is fine.
